### PR TITLE
Validate tool activities

### PR DESCRIPTION
**Pull request by collindutter**
Addresses https://github.com/griptape-ai/griptape/issues/128

### CODE REVIEW

File: griptape/core/activity_mixin.py [hunk 1]
```diff
@@ -12,14 +12,26 @@ class ActivityMixin:
...
+        if self.denylist is not None:
             raise ValueError("can't have both allowlist and denylist specified")
 
+        for activity_name in allowlist:
+            self._validate_tool_activity(self.__class__, activity_name)
+
     @denylist.validator
-    def validate_allowlist(self, _, denylist: Optional[list[str]]) -> None:
-        if self.allowlist is not None and denylist is not None:
+    def validate_denylist(self, _, denylist: Optional[list[str]]) -> None:
```
Comment (collindutter) [inline griptape/core/activity_mixin.py:25]: Unrelated change, method name was improperly named.

**Review by vasinov — APPROVED** [review]
